cli/src/utils/command.ts

```diff
@@ -0,0 +1,114 @@
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import { URL } from 'url';
+import { Command } from '@oclif/command';
+import { CLIError } from '@oclif/errors';
+import * as Parser from '@oclif/parser';
+import open from 'open';
+import ms from 'ms';
+import terminalLink from 'terminal-link';
+import { LycanClient } from '@binaris/spice-node-client';
+import flags from './cli-flags';
+import userConfig from './user-config';
+
+const pjson = JSON.parse(readFileSync(resolve(__dirname, '../../package.json'), 'utf-8'));
+
+const LOGIN_PARAM = 'ticket';
+const TICKET_CLAIM_INTERVAL_MS = 1000;
+
+export default abstract class BaseCommand extends Command {
+  static cliBinName = pjson.oclif.bin as string;
+
+  protected realm?: string;
+  protected lycanClient?: LycanClient;
+
+  public static flags = {
+    help: flags.help({ char: 'h' }),
+    realm: flags.string({
+      default: 'prod',
+      hidden: true,
+      env: 'realm',
+    }),
+  };
+
+  protected parse<F, A extends {[name: string]: any}>(opt?: Parser.Input<F>, argv = this.argv): Parser.Output<F, A> {
+    if (!opt) {
+      opt = this.constructor as any;
+    }
+    if (!opt!.flags) {
+      opt!.flags = BaseCommand.flags as any;
+    }
+
+    return require('@oclif/parser').parse(argv, { context: this, ...opt });
+  }
+
+  public async init() {
+    const { flags: { realm } } = this.parse(BaseCommand);
+    this.realm = realm;
+    this.lycanClient = new LycanClient(`${this.getBaseUrl('api')}/public/v1`);
+  }
+
+  public async authenticate(force = false): Promise<string> {
```

Comment:
`force` is not a good name, can we call it `useLocalToken` or `skipLocalToken` or `remoteOnly`?

Reply:
`useLocalToken` - I don't like it because you usually use it.
`skipLocalToken` - Not a good name IMO because the new token will be saved
`remoteOnly` - Doesn't say anything to me

do you have a problem with the world `force`?
how about 'forceBrowserAuthFlow' ?

Reply:
forceBrowserAuthFlow is good, thanks!

Reply:
done